CenterForOpenScience/osf.io
# [ENG-2799] Schema add YOUth study registration template

## Purpose

This adds a new type of schema and a new instance of that schema to our current schema system. The new type of schema is called `simpleSchema` and it more closely mirrors the schema blocks model we currently use.

A single "block" in the single schema is simply a JSON blob of that SchemaBlocks attributes, for example:
```
{
"help_text": "",
 "example_text": "",
 "block_type": "long-text-input",
 "display_text": "",
 "required": false
}
```

is now the block for a textarea.

## Changes

- add custom `simpleSchema` reading logic in `map_to_schema_blocks`
- add teeny tiny migration so SchemaBlocks will allow some null fields


## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify the schemas are correct 


## Documentation

🐞 fix, no docs

## Side Effects

None that I know of.


## Ticket

https://openscience.atlassian.net/browse/ENG-2799

# Preview: Hypothesis-testing studies using YOUth data

![page1](https://user-images.githubusercontent.com/9688518/122228212-a0485800-ce85-11eb-88d3-8b92b9801a8b.png)
![page2](https://user-images.githubusercontent.com/9688518/122228209-a0485800-ce85-11eb-8b03-b7f6290d58ad.png)
![page3](https://user-images.githubusercontent.com/9688518/122228205-9fafc180-ce85-11eb-8dd1-2afe3487e44b.png)
![page4](https://user-images.githubusercontent.com/9688518/122228204-9f172b00-ce85-11eb-95fe-6e8fb0d13571.png)
![page5](https://user-images.githubusercontent.com/9688518/122228199-9e7e9480-ce85-11eb-8f48-ca1ba9784a31.png)

# Preview: Other studies using YOUth data

![page1-2](https://user-images.githubusercontent.com/9688518/122242329-3b92fa80-ce91-11eb-81d0-206de50f1c57.png)
